Updated singular tests section to show they can now be documented (#6147)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/data-tests.md

The name of this test is the name of the file: `assert_total_payment_amount_is_positive`. Simple enough.

Singular data tests are easy to write—so easy that you may find yourself writing the same basic structure over and over, only changing the name of a column or model. By that point, the test isn't so singular! In that case, we recommend [generic data tests](#generic-data-tests).

### Document singular data tests

<VersionBlock firstVersion="1.9">

The top-level `data_tests:` key in YAML files allows for adding a [`description`](/reference/resource-properties/description) for singular data tests. Descriptions can include Markdown, as well as the [`doc` jinja function](/reference/dbt-jinja-functions/doc).

<File name='tests/singular/data_tests.yml'>

```yml

# top-level
data_tests:
  - name: my_super_cool_singular_test
    description: The `order_id` is unique for every row in the orders model
```

</File>

</VersionBlock>

<VersionBlock lastVersion="1.8">

> The `description` property is available for [singular data tests](/build/data-tests#singular-data-tests) beginning in dbt v1.9.

Reviewer: ```suggestion
The `description` property is available for [singular data tests](/build/data-tests#document-singular-data-tests) beginning in dbt v1.9.
```

Singular data tests are easy to write—so easy that you may find yourself writing the same basic structure over and over, only changing the name of a column or model. By that point, the test isn't so singular! In that case, we recommend...

<VersionBlock firstVersion="1.9">

### Document singular tests

You can set descriptions in your YAML files to document singular data tests:

> <File name='data_tests.yml'>

Reviewer 2: While any of the [`docs-paths`](https://docs.getdbt.com/reference/project-configs/docs-paths) will work, the `tests` directory makes sense here because that is the default for [`test-paths`](https://docs.getdbt.com/reference/project-configs/test-paths). Furthermore, using `singular` not because it is required (it's not!), but because it is a very readable and logical complement to the `generic` subdirectory that contains definitions for custom generic data tests.

```suggestion
<File name='tests/singular/data_tests.yml'>
```

> Previously, documenting a data test such as a singular data test would result in a parsing error. However, in [Versionless](/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can set descriptions in your YAML files to document singular data tests.

Reviewer 3: I don't think we need to callout previous behavior. Instead, a callout that this is available in versionless and will be available in dbt Core `v1.9` and then just a description of the feature.

> Previously, documenting a data test such as a singular data test would result in a parsing error. However, in [Versionless](/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can set descriptions in your YAML files to document singular data tests.

Reviewer 2: ```suggestion
You can set descriptions in your YAML files to document singular data tests:
```

> Previously, documenting a data test such as a singular data test would result a parsing error. However, in [Versionless](/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can allow setting descriptions in your yaml files to enable you to document singular data tests.

Reviewer: ```suggestion
Previously, documenting a data test such as a singular data test would result in a parsing error. However, in [Versionless](/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can set descriptions in your YAML files to document singular data tests.
```

```yml

# top-level
data_tests:
  - name: my_super_cool_singular_test
    description: The order_id is unique for every row in the orders model
    config:
      error_if: ">10"
  - name: my_super_cool_singular_test2

>     description: '{{ doc("not_null_test") }}'

Reviewer 3: @aranke Can you confirm that docs blocks will work for documenting singular tests as well?

Reviewer 2: ## TLDR
We have two issues open for adding `description` to data tests:
- https://github.com/dbt-labs/dbt-core/issues/9005#issuecomment-1834047344
- https://github.com/dbt-labs/dbt-core/issues/2578

This PR _only_ allows the first and not the second.

## Details

There are two types of data tests:
1. Generic
1. Singular

### Descriptions for generic data tests

Generic data tests have one macro _definition_ and then zero to many _instances_ where they are used. So that's two different places we can imagine adding a description for generic data tests. My understanding is the first is supported by this PR but the second is **not**.

Since the _definition_ is just a macro, the definition of a custom generic data test can already have a [`description`](https://docs.getdbt.com/reference/resource-properties/description) like this:

```yaml
macros:
  - name: test_CUSTOM_GENERIC_TEST_NAME
    description: markdown_string
    arguments:
      - name: argument_name
        description: markdown_string
```

However, I don't believe the _instances_ of generic data tests can be documented until https://github.com/dbt-labs/dbt-core/issues/2578 is implemented.

### Descriptions for singular data tests

Added in https://github.com/dbt-labs/dbt-core/pull/10792 to resolve https://github.com/dbt-labs/dbt-core/issues/9005!

I will test out this example to make sure it works 👍

Reviewer 2: ✅ Works
